SoftLayer/managers/cdn.py

```diff
@@ -170,3 +174,89 @@ def start_data(self):
     def end_date(self):
         """Retrieve the cdn usage metric end date."""
         return self._end_date
+
+    def edit(self, identifier, header=None, http_port=None, origin=None,
+             respect_headers=None, cache=None, performance_configuration=None):
+        """Edit the cdn object.
+
+        :param string identifier: The CDN identifier.
+        :param header: The cdn Host header.
+        :param http_port: The cdn HTTP port.
+        :param origin: The cdn Origin server address.
+        :param respect_headers: The cdn Respect headers.
+        :param cache: The cdn Cache key optimization.
+        :param performance_configuration: The cdn performance configuration.
+
+        :returns: SoftLayer_Container_Network_CdnMarketplace_Configuration_Mapping[].
+        """
+        cdn_instance_detail = self.get_cdn(str(identifier))
+
+        config = {
+            'uniqueId': cdn_instance_detail.get('uniqueId'),
+            'originType': cdn_instance_detail.get('originType'),
+            'protocol': cdn_instance_detail.get('protocol'),
+            'path': cdn_instance_detail.get('path'),
+            'vendorName': cdn_instance_detail.get('vendorName'),
+            'cname': cdn_instance_detail.get('cname'),
+            'domain': cdn_instance_detail.get('domain'),
+            'httpPort': cdn_instance_detail.get('httpPort'),
+            'origin': cdn_instance_detail.get('originHost')
```

Comment:
```suggestion
            'origin': cdn_instance_detail.get('originHost')
            'header': cdn_instance_detail.get('header')
```
Please add `header` as a default value too.

Reply:
What is the purpose of adding the `header` data here for updating the cdn object?

Reply:
if the` header` is here, this one just will update if is passed as explicit param. so the header will behavior, in the same way, the `origin`, and the `httpPort` are working.

Reply:
Done, thanks.